system/jlib/jmutex.hpp

```diff
@@ -411,7 +413,17 @@ class jlib_decl  SpinLock
             return;
         }
         while (unlikely(!atomic_acquire(&value)))
-            ThreadYield();
+        {
+            unsigned i = 0;
```

Comment:
I think definitely - I just looked at the generated code, and it has unrolled the loop, so each use is about 100bytes bigger.  That could be painful.

A global function like

void jlib_decl spinUntilReady(atomic_t & value);

should work.